### PR TITLE
Add invoice object

### PR DESCRIPTION
**Pull request by zhenlu**

### CODE REVIEW

File: uma/protocol/invoice.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+from uma.protocol.counterparty_data import (
+    CounterpartyDataOption,
+    CounterpartyDataOptions,
+)
+from uma.protocol.kyc_status import KycStatus
+from uma.tlv_utils import TLVCodable, ByteCodable
+
+
+@dataclass
+class InvoiceCurrency(TLVCodable):
```
Comment (jklein24) [inline uma/protocol/invoice.py:15]: Do we also need `decimals` here for display purposes on the sending side?

File: uma/protocol/invoice.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+        for pair in data.decode().split(","):
+            key, value = pair.split(":")
+            options[key] = CounterpartyDataOption(mandatory=value == "1")
+        return cls(options)
+
+
+@dataclass
+class Invoice(TLVCodable):
+    # Receiving UMA address
+    receiverUma: str
```
Comment (jklein24) [inline uma/protocol/invoice.py:68]: Fields should probably be snake_case in python. Is there a way we can do that without making the encoding/decoding a huge pain? We do that with the JSONable class in this sdk and the lightspark-sdk for example.

File: uma/protocol/invoice.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    # containing the lowest minor version of each major version it supported, separated by commas.
+    umaVersion: str
+
+    # CommentCharsAllowed is the number of characters that the sender can include in the comment field of the pay request.
+    commentCharsAllowed: Optional[int]
+
+    # The sender's UMA address. If this field presents, the UMA invoice should directly go to the sending VASP instead of showing in other formats.
+    senderUma: Optional[str]
+
+    # The maximum number of the invoice can be paid
```
Comment (jklein24) [inline uma/protocol/invoice.py:98]:
```suggestion
    # The maximum number of times the invoice can be paid
```

File: uma/protocol/invoice.py (hunk 2 above)
Comment (zhenlu) [reply uma/protocol/invoice.py:68]: Hmm it shouldn't be a problem as the map function maps the field and tlv type. But I'm surprised that pyre, pylint and black didn't warn me about the snake case after all the trouble I had with them :old-man-yells-at-python:

**Review by jklein24 — APPROVED** [review]

**Pull request merged**